`attributes` rule gives false error in protocol implementation after @testable

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

`attributes` rules gives a false error on Xcode 9.3: 

```
File.swift:15:15: warning: Attributes Violation: Attributes should be on their own lines in functions and types, but on the same line as variables and imports. (attributes)
```

Code:

```
import SomeFramework

@testable import SomeApp

private final class SomeClass: SomeProtocol { // error here
    ...
}
```


### Environment

* SwiftLint version: 0.25.0
* Installation method used: cocoapods
* Paste your configuration file:

```yml
attributes:
  always_on_line_above:
    - "@objc"
    - "@nonobjc"
    - "@available"
    - "@discardableResult"
  always_on_same_line:
    - "@IBOutlet"
    - "@IBAction"
    - "@NSManaged"
    - "@testable"
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? Yes, one at the root folder and one in `AppTests`.

* Which Xcode version are you using (check `xcode-select -p`)? Xcode 9.3 9E145
* Do you have a sample that shows the issue? Yes, see above

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -92,6 +92,11 @@
   even if only 1 file changed.  
   [John Szumski](https://github.com/jszumski)
 
+* Fix false positives in `attributes` rule when using Swift 4.1.  
+  [Marcelo Fabri](https://github.com/marcelofabri)
+  [#2125](https://github.com/realm/SwiftLint/issues/2125)
+  [#2141](https://github.com/realm/SwiftLint/issues/2141)
+
 ## 0.25.0: Cleaning the Lint Filter
 
 #### Breaking

--- a/Rules.md
+++ b/Rules.md
@@ -412,6 +412,16 @@ func increase(f: @autoclosure () -> Int) -> Int
 func foo(completionHandler: @escaping () -> Void)
 ```
 
+```swift
+private struct DefaultError: Error {}
+```
+
+```swift
+@testable import foo
+
+private let bar = 1
+```
+
 </details>
 <details>
 <summary>Triggering Examples</summary>

--- a/Source/SwiftLintFramework/Rules/AttributesRule.swift
+++ b/Source/SwiftLintFramework/Rules/AttributesRule.swift
@@ -311,7 +311,17 @@ public struct AttributesRule: ASTRule, OptInRule, ConfigurationProviderRule {
             "source.decl.attribute.postfix",
             "source.decl.attribute.prefix",
             "source.decl.attribute.required",
-            "source.decl.attribute.weak"
+            "source.decl.attribute.weak",
+            "source.decl.attribute.private",
+            "source.decl.attribute.fileprivate",
+            "source.decl.attribute.internal",
+            "source.decl.attribute.public",
+            "source.decl.attribute.open",
+            "source.decl.attribute.setter_access.private",
+            "source.decl.attribute.setter_access.fileprivate",
+            "source.decl.attribute.setter_access.internal",
+            "source.decl.attribute.setter_access.public",
+            "source.decl.attribute.setter_access.open"
         ]
         return attributes.filter { !blacklist.contains($0) }
     }

--- a/Source/SwiftLintFramework/Rules/AttributesRulesExamples.swift
+++ b/Source/SwiftLintFramework/Rules/AttributesRulesExamples.swift
@@ -44,7 +44,9 @@ internal struct AttributesRuleExamples {
         "@discardableResult\n func a() -> Int",
         "@objc\n @discardableResult\n func a() -> Int",
         "func increase(f: @autoclosure () -> Int) -> Int",
-        "func foo(completionHandler: @escaping () -> Void)"
+        "func foo(completionHandler: @escaping () -> Void)",
+        "private struct DefaultError: Error {}",
+        "@testable import foo\n\nprivate let bar = 1"
     ]
 
     static let triggeringExamples = [

